### PR TITLE
[CST] - Closed claim status tab changes for V2

### PR DESCRIPTION
**Pull request by pmclaren19**
## Summary

Deleted CompleteDetails.jsx and created a Payments.jsx component 
- Updated ClaimComplete.jsx and replaced the CompleteDetails component with paragraph text and the Payment component
- Update ClaimTimeline.jsx and replaced the CompleteDetails component with the html from CompleteDetails
- Update ClaimTimelines.jsx and replaced the CompleteDetails component with the html from CompleteDetails

Updated ClaimStatusHeader.jsx so that it uses the helper isClaimOpen and the Last Updated Date only displays when the claim is opened

Created NextSteps.jsx that replaces the next steps section
- New component is used on ClaimStatusPage.jsx and ClaimsDecision.jsx

Created ClosedClaimAlert.jsx thats displayed on the Status Tab page when a claim is closed.

Created Payments.jsx that replaces the payments section
- New component is used on ClaimStatusPage.jsx and ClaimComplete.jsx

Created a new folder called `claim-status-tab` in components and in tests to hold all of the status tab components and tests. Added the following components to it and updated their imports.
- ClosedClaimAlert
- NextSteps
- Payments
- RecentActivity
- WhatWeAreDoing
- WhatYouNeedToDo

Moved the ClaimsTimeline.jsx to the evss folder since this is only used for evss purposes. Also removed the CompleteDetails component and replaced with the html 

Added new const and functions to the helpers.js for getStatusMap and isClaimOpen
- Added these function to ClaimStatusHeader.jsx, OverviewPage.jsx, FilesPage.jsx and ClaimStatusPage.jsx 

Updated the following tests:
- src/applications/claims-status/tests/components/ClaimStatusHeader.unit.spec.jsx
- src/applications/claims-status/tests/components/ClaimComplete.unit.spec.jsx - changed to not use SkinDeep
- src/applications/claims-status/tests/components/ClaimsDecision.unit.spec.jsx

Added test files for the following:
- src/applications/claims-status/tests/components/claim-status-tab/ClosedClaimAlert.unit.spec.jsx
- src/applications/claims-status/tests/utils/helpers.unit.spec.js
- src/applications/claims-status/tests/components/claim-status-tab/NextSteps.unit.spec.jsx
- src/applications/claims-status/tests/components/claim-status-tab/Payments.unit.spec.jsx

## Tasks
 
- [x] An information alert displays with text and a link when a claim is closed with a decision letter sent
- [x] An information alert displays with text when a claim is closed with no decision letter sent
- [x] Add logic information alert to account for a null closed date
- [x] A "Payments" section is created and is only shown when a claim is closed
- [x] The "Last updated date" is hidden when claim is closed
- [x] The "Next Steps" section is shown when claim is closed
- [x] The "Recent Activity" section is shown when claim is closed
- [x] The "What you need to do" section is hidden when claim is closed
- [x] The "What we're doing" section is hidden when claim is closed
- [x] The "Claim Complete" component isn't displayed when cstUseClaimDetailsV2 feature toggle is enabled
- [x] The "Claims Decision" component isn't displayed when cstUseClaimDetailsV2 feature toggle is enabled
- [x] Design Check

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#77239

## Testing done

- _Describe what the old behavior was prior to the change_
- _Describe the steps required to verify your changes are working as expected_
- _Describe the tests completed and the results_
- _Exclusively stating 'Specs and automated tests passing' is NOT acceptable as appropriate testing
- _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Status Tab - Closed Claim - decisionLetterSent - Toggle Off | Screenshot 2024-02-29 at 1 45 26 PM | Screenshot 2024-02-29 at 3 55 05 PM |
| Status Tab - Closed Claim - no decisionLetterSent - Toggle Off | Screenshot 2024-02-29 at 4 09 31 PM | Screenshot 2024-02-29 at 4 06 19 PM |
| Status Tab - Closed Claim - decisionLetterSent - Toggle On | Screenshot 2024-02-29 at 4 12 42 PM | Screenshot 2024-02-29 at 4 16 34 PM |
| Status Tab - Closed Claim - no decisionLetterSent - Toggle On | Screenshot 2024-02-29 at 4 14 17 PM | Screenshot 2024-02-29 at 4 17 24 PM |
| Status Tab - Closed Claim - null closed date - Toggle Off | Screenshot 2024-03-01 at 4 15 58 PM | N/A |
| Status Tab - Closed Claim - null closed date - Toggle On | N/A | Screenshot 2024-03-01 at 4 18 44 PM |
| Status Tab - Opened Claim - Toggle On | Screenshot 2024-03-04 at 8 41 01 AM | Screenshot 2024-03-04 at 8 39 52 AM |
| Status Tab - Opened Claim - Toggle Off | Screenshot 2024-03-04 at 8 42 48 AM | Screenshot 2024-03-04 at 8 51 24 AM |
## What areas of the site does it impact?

Claim Status Tool

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Comment by pmclaren19** [comment]
Met with @skylerschain on Friday and we did a Design Check!

File: src/applications/claims-status/tests/components/ClaimComplete.unit.spec.jsx [hunk 1]
```diff
@@ -1,17 +1,22 @@
...
-    expect(tree.text()).to.contain(
-      'We decided your claim on September 1, 2010',
+    const { container, getByText } = render(
+      <ClaimComplete completedDate={date} />,
     );
-    expect(tree.everySubTree('CompleteDetails')).to.not.be.false;
+    const paymentsSection = $('.payments-container', container);
+    expect(paymentsSection).to.exist;
+
+    const text1 =
```
Comment (samcoforma) [inline src/applications/claims-status/tests/components/ClaimComplete.unit.spec.jsx:17]: Is there a text2?

Comment (pmclaren19) [reply src/applications/claims-status/tests/components/ClaimComplete.unit.spec.jsx]: no there is not! Updated to just be text @samcoforma

File: src/applications/claims-status/tests/utils/helpers.unit.spec.js [hunk 2]
```diff
@@ -662,6 +687,42 @@ describe('Disability benefits helpers: ', () => {
...
+
+      context('when closeDate exists', () => {
+        it('should return false', () => {
+          const isOpen = isClaimOpen(status, '2024-01-01');
+          expect(isOpen).to.be.false;
+        });
+      });
+    });
+
+    context('when status is not COMPELTE', () => {
```
Comment (samcoforma) [inline src/applications/claims-status/tests/utils/helpers.unit.spec.js:708]: typo here: "COMPELTE" should be "COMPLETE"

Comment (pmclaren19) [reply src/applications/claims-status/tests/utils/helpers.unit.spec.js:708]: updated thanks!

File: src/applications/claims-status/tests/utils/helpers.unit.spec.js [hunk 3]
```diff
@@ -662,6 +687,42 @@ describe('Disability benefits helpers: ', () => {
     });
   });
 
+  describe('isClaimOpen', () => {
+    context('when status is COMPELTE', () => {
```
[the remark this answers is not in the document]
Comment (pmclaren19) [reply src/applications/claims-status/tests/utils/helpers.unit.spec.js:691]: updated thanks!

**Review by samcoforma — APPROVED** [review]

**Pull request merged**